anoncreds/src/services/prover.rs

```diff
@@ -46,6 +48,10 @@ pub fn create_credential_request(
         credential_offer
     );
 
+    let rand_str = String::from_utf8(thread_rng().sample_iter(&Alphanumeric).take(22).collect())
```

Comment:
The value prover_did should become optional in the whole credential request. If not present in the request, the issuer will generate it. 

So if no prover_did is present we don't include it in the request